src/main/java/de/tum/in/www1/artemis/web/rest/QuizExerciseResource.java

```diff
@@ -196,13 +214,15 @@ public ResponseEntity<QuizExercise> updateQuizExercise(@RequestBody QuizExercise
 
         // Valid exercises have set either a course or an exerciseGroup
         quizExercise.checkCourseAndExerciseGroupExclusivity(ENTITY_NAME);
+
+        final var originalQuiz = quizExerciseRepository.findWithEagerQuestionsById(exerciseId).orElseThrow();
```

Comment:
Consider adding a `findWithEagerQuestionsByIdOrElseThrow()` method to the repository, you are currently throwing a custom exception below but not here.